third-party/realdds/include/realdds/dds-stream-base.h

```diff
@@ -0,0 +1,55 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+
+#include "dds-stream-profile.h"
+
+#include <memory>
+#include <string>
+
+
+namespace realdds {
+
+
+class dds_topic;
+
+
+// Base class for both client/subscriber and server/publisher stream implementations: contains
+// information needed to identify a stream, its properties, and its profiles.
+//
+class dds_stream_base : public std::enable_shared_from_this< dds_stream_base >
+{
+protected:
+    std::string const _name;
+    std::string const _sensor_name;
```

Comment:
Do we want to mention `sensor`? Call it `_group_name`

Reply:
I was debating and in the end thought that "group" was too ambiguous. Didn't get to talk to you.
I'm leaning away from "group" -- if this is realdds, for realsense, then we could use "sensor" as long as we don't center things around it.

Reply:
I'm OK with this. Group was more generic and suited applications other than LibRS but that is not the scope now